gateway-spi/src/main/java/org/apache/knox/gateway/dispatch/DefaultHttpClientFactory.java

```diff
@@ -128,12 +128,14 @@ public HttpClient createHttpClient(FilterConfig filterConfig) {
 
     if (doesRetryParamExist(filterConfig)) {
       int retryCount = Integer.parseInt(filterConfig.getInitParameter(PARAMETER_RETRY_COUNT));
+      LOG.setRetryCount(retryCount, serviceRole);
       /* do we want to retry non-idempotent requests? default no */
       boolean retryNonIdempotent = DEFAULT_PARAMETER_RETRY_NON_SAFE_REQUEST;
       if (filterConfig.getInitParameter(PARAMETER_RETRY_NON_SAFE_REQUEST)
           != null) {
         retryNonIdempotent = Boolean.parseBoolean(
             filterConfig.getInitParameter(PARAMETER_RETRY_NON_SAFE_REQUEST));
+        LOG.setRetryNonIndependent(retryNonIdempotent, serviceRole);
```

Comment:
IMO, this should be outside of this `if` statement. Otherwise, we would assume it's set to default because the log entry was not shown.